File: app/src/main/java/gallery/memories/service/DownloadBroadcastReceiver.java
package gallery.memories.service;

import android.app.DownloadManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import gallery.memories.NativeX;

public class DownloadBroadcastReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (NativeX.mDlService != null) {
            NativeX.mDlService.runDownloadCallback(intent);
        }
    }
}

File: app/src/main/java/gallery/memories/service/DownloadBroadcastReceiver.kt
package gallery.memories.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.media3.common.util.UnstableApi
import gallery.memories.NativeX

@UnstableApi class DownloadBroadcastReceiver : BroadcastReceiver() {
    /**
     * Callback when download is complete
     */
    override fun onReceive(context: Context, intent: Intent) {
        NativeX.dlService?.runDownloadCallback(intent)
    }
}